homeworks/rostyslav.havuka_RostyslavHavuka/6-typeScript/src/index.ts

```diff
@@ -0,0 +1,239 @@
+const scoreElement = document.querySelector('[data-score]') as HTMLHeadingElement;
+const cubeElement = document.querySelector('[data-cube]') as HTMLDivElement;
+const cubeScoreElement = document.querySelector('[data-cube-score]') as HTMLDivElement;
+const keyElement = document.querySelector('[data-key]') as HTMLInputElement;
+const progressBarElement = document.querySelector('[data-progress-bar]') as HTMLElement;
+const keyStart = document.querySelector('[data-key-start]') as HTMLButtonElement;
+const keyStop = document.querySelector('[data-key-stop]') as HTMLButtonElement;
+const keyAgain = document.querySelector('[data-key-again]') as HTMLButtonElement;
+const square = document.getElementById("box") as HTMLElement;
+
+class Game {
+  private currentKey: string = '';
+  private score: number = 100;
+  private intervalId: NodeJS.Timeout;
+  private intervalStatusBarId: NodeJS.Timeout;
+  private gameState: boolean = false;
+  private timerProgressBar: number = 0;
+  private squareWidth: number;
+
+  constructor(
+    private scoreElement: HTMLHeadingElement,
+    private cubeScoreElement: HTMLDivElement,
+    private keyElement: HTMLInputElement,
+    private cubeElement?: HTMLDivElement,
+    private progressBarElement?: HTMLElement,
+    private square?: HTMLElement,
+    private keyStart?: HTMLButtonElement,
+    private keyStop?: HTMLButtonElement,
+    private keyAgain?: HTMLButtonElement
+  ) { }
+
+  start() {
+    this.randomKeyInterval()
+
+    this.subscribeOnKeyPress()
+
+    this.keyStart.addEventListener('click', (): void => {
+      this.subscribeOnKeyStartClick();
+    })
```

Comment:
Not quite what I meant. `this.subscribeOnKeyStartClick` method should include the event listener. **And you need to call this function once in the start method.**

Applicable to all event listeners below.